src/vcStringFormat.cpp

```diff
@@ -26,25 +25,27 @@ const char *vcStringFormat(const char *pFormatString, const char **ppStrings, si
 
       int length;
       currArg = udStrAtoi((pFormatString + i + 1), &length);
-      if (currArg > (int) numStrings || currArg < 0 || pFormatString[length + i + 1] != '}')
+      if (currArg > numStrings || pFormatString[length + i + 1] != '}')
```

Comment:
It's very confusing to see `udStrAtoi` and the removal of the `< 0` check. I'm genuinely surprised that this doesn't generate any warnings, `currArg` is unsigned but `udStrAtoi` is returning a signed value. You should really change all of the `udStrAtoi` usages to `udStrAtou` for clarity.